### ISSUE TITLE
Cannot access property started with '\0'

### ISSUE THREAD

**Issue by mircobabini** [deliciousbrains/better-search-replace#71]
`PHP Fatal error:  Uncaught Error: Cannot access property started with '\0' in /wp-content/plugins/better-search-replace/includes/class-bsr-db.php:344`

It's a known issue on PHP 7.3.
This is because of the empty char which IS NOT exactly an empty character but "*_".
You can't have a property starting with *.

$item->*_tbl becomes $item->\0_tbl. It's from the wp_options in my case.

**Btw, solution is:**
```
$key = trim( $key );
$this->$key = ...
```

**Comment by mapsmarketing** [comment]
Thank you very much @mircobabini

**Comment by jdreesen** [comment]
Sounds more like you're dealing with the array representation of a protected property, whose key would be of format `\0*\0property_name`. Just `trim`ming the `\0` wouldn't help here, because:

> You can't have a property starting with *

Yes, you can if you set it like this: https://3v4l.org/P81e4, which is exactly what your proposed fix is doing. 
What you *can't* have is a property starting with `\0`, though, which is why your fix *seems* to work.
But this wouldn't help, because it's a different name that's nobody going to use (also: this is [deprecated in PHP 8.2](https://www.php.net/releases/8.2/en.php#deprecate_dynamic_properties)).

My proposal would be to skip those properties via:

```php
if (!str_starts_with($key , "\0")) {
    $_tmp->$key = ...
}
```

**Comment by AhmedTheGeek** [comment]
Thanks for the report @mircobabini, and the fix suggestion @jdreesen. This issue has been resolved by #101.
